### ISSUE TITLE
$-expressions are not evaluated on client/master in distributed tests

### ISSUE THREAD

**Issue by markuswege** [apache/jmeter#5896]
### Expected behavior

In distributed testing ("master/slave") the evaluation of $-expressions works in ResultsCollector, BackendListener and User Defined Variables.

### Actual behavior

In distributed testing ("master/slave") the evaluation of $-expressions works in ResultsCollector, BackendListener but **not** User Defined Variables.

### Steps to reproduce the problem

Create simple Test having User Defined Variables and BackendListener.

User Defined Variables:
`TEST_PROFILE=${__P(activeProfileId)}`

BackendListener:
`profileId=${TEST_PROFILE}`

Run in distributed setup.

Client (master) does not evaluate $-expressions in User Defined Variables.
In the given case, the content of profileId is "`${__P(activeProfileId)}`" instead of the value passed by the property "`activeProfileId`".

Note: We already implemented a fix for this, Pull Request will follow soon.

### JMeter Version

5.5

### Java Version

OpenJDK Runtime Environment 18.9 (build 11+28)

### OS Version

openjdk version "11" 2018-09-25

**Comment by vlsi** [comment]
This might duplicate https://github.com/apache/jmeter/issues/5294

**Comment by markuswege** [comment]
Yes,  pretty mich looks like it. Will keep this one open nevertheless since i hace fixed it.
Tests are WiP, since there are no tests for the classes involved, i need to start from scratch here.
